core/src/main/scala/io/finch/Endpoint.scala

```diff
@@ -95,6 +95,24 @@ trait Endpoint[F[_], A] { self =>
       final override def toString: String = self.toString
     }
 
+  /**
+    * Maps this endpoint to the given function `F[A] => F[B]`
+    */
+  final def mapF[B](fn: F[A] => F[B])(implicit F: Monad[F]): Endpoint[F, B] =
```

Comment:
Is it is a common name for this kind of combinator?

We already have `transform(f: F[Output[A]] => F[Output[B]])` that looks fairly similar to this. Maybe we should use symmetric name here as well?

I feel like it should be 

 - `transform(f: F[A] => F[B])`
 - `transformOutput(f: F[Output[A]] => F[Output[B]])`

to match the current naming schema. I'm open to run this through a deprecation cycle and fix the names.

Reply:
I followed Kleisli naming here:
https://github.com/typelevel/cats/blob/89b13d6d7c0933fe5639f639a89745dcf571fc1b/core/src/main/scala/cats/data/Kleisli.scala#L33

But `transform` is also a common name there.

Reply:
Linked `mapF` from Kleisli is different though, it maps over an effect type, `F[_] = G[_]`.

What do you think about deprecating `transform` in this release and renaming to `transformOutput` so we can introduce a proper transform in the next milestone?

Reply:
not really a map over the _type_ but also over the value `F[B] => N[C]`.
`mapK` is for the transformation `F ~> G`

This one is actually quite important in our current projects. Could we somehow speed it up?

Ofc we could make it on our own using implicits for now.

Reply:
Sure. What about `transformF` for this release? We can do the renaming/deprecations later.

Reply:
Done